tests/test_apply.py

```diff
@@ -1135,7 +1134,7 @@ class TestTensorSample:
 
     def test_paulix_pauliy(self, theta, phi, varphi, monkeypatch, tol):
         """Test that a tensor product involving PauliX and PauliY works correctly"""
-        dev = qml.device("lightning.qubit", wires=3, shots=10000)
+        dev = qml.device("lightning.qubit", wires=3, shots=None)
```

Comment:
Why does this one need to change?

Reply:
Because we implicitly had `analytic=True` before

Reply:
:thinking: This one's still not clear to me - so before we had `analytic=True` and `shots=10000`. In the test, we call `dev.sample()` so it would have generated `10000` samples, yet these samples were not used in `dev.probability()` because we were in analytic mode?

So perhaps the test didn't need to set `shots` or call `dev.sample()`?

Would this just work if we had a parametrize over shots, setting `[None, 10000]`, so we test both analytic and non-analytic cases?

Reply:
Sure, changed this. Also removed the calls to `dev.sample`.